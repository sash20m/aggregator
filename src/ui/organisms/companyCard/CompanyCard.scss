.results {
    display: flex;
    width: 100%;
    flex-direction: column;
    align-items: center;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

    &__item {
        display: flex;
        width: 85%;
        border-radius: 5px;
        flex-direction: column;
        margin: 20px 0 20px 0;
        border: 1px solid #e0e5ed;

        &:hover {
            display: flex;
            width: 85%;
            border: 1px solid #1890ff;
            border-radius: 5px;
            flex-direction: column;
            margin: 20px 0 20px 0;
            -webkit-box-shadow: 4px 4px 39px -22px rgba(63, 65, 66, 0.35);
            -moz-box-shadow: 4px 4px 39px -22px rgba(63, 65, 66, 0.35);
            box-shadow: 4px 4px 39px -22px rgba(63, 65, 66, 0.35);
        }

        &__header {
            display: flex;
            flex-direction: row;
            width: 95%;
            margin-top: 30px;
            margin-bottom: 30px;
            margin-left: 35px;
            outline: 0;

            &:hover {
                cursor: pointer;
            }


            &__avatar {
                width: 5em;
            }

            &__name {
                font-size: 1.3em;
                color: #467bbd;
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                font-weight: 700;
                margin: 17px 0 7px 27px;
            }

            &__location {
                font-size: 1;
                color: rgba(0, 0, 0, 0.65);
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                font-weight: 700;
                margin-left: 27px;
            }
        }

        &__company-info {
            display: flex;
            flex-direction: row;
            width: 95%;
            margin-left: 35px;

            &__column {

                &__item {
                    display: flex;
                    margin: 0 150px 15px 0;
                    font-size: 0.9em;
                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

                    &--key {
                        color: rgba(0, 0, 0, 0.40);
                        margin: 0 5px 0 0;
                    }

                    &--value {
                        font-weight: 600;
                        margin: 0;
                    }
                }
            }

            &__founder {
                display: flex;
                flex-direction: row;
                width: 95%;
                border-top: 1px solid rgb(223, 223, 223);
                margin-left: 35px;
                padding: 10px 0 10px 0;

                &--key {
                    color: rgba(0, 0, 0, 0.40);
                    margin: 0 5px 0 0;
                }

                &--value {
                    font-weight: 600;
                    margin: 0;
                    margin-left: 10px;
                }
            }
        }

    }

}