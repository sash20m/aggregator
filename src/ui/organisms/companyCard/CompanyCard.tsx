import React from 'react';

import Link from 'next/link';
import { CompaniesCard } from 'ui/templates/companies/Companies';

import './CompanyCard.scss';

interface Props {
  data: CompaniesCard;
}

export const CompanyCard: React.FC<Props> = ({ data }) => {
  // const Router = useRouter();
  // const scrollRef = useRef<HTMLDivElement>(null);
  const yearsOld = new Date().getFullYear() - data.creation_year;

  let imageURL = '';
  if (data.website) {
    imageURL = new URL(data.website).hostname;
  }

  return (
    <div className="results">
      <div className="results__item">
        <Link
          href="/company/[slug]"
          as={`/company/${data.slug}`}
          data-testid="link"
        >
          <button
            data-testid={`${data.slug}`}
            className="results__item__header"
            type="button"
          >
            {imageURL ? (
              <img
                className="results__item__header__avatar"
                src={`https://account.globaldatabase.com/logo/${imageURL}/`}
                alt=""
              />
            ) : (
              <div className="results__item__header__no-avatar">
                {data.name[0]}
              </div>
            )}
            <div>
              <div className="results__item__header__name">{data.name}</div>
              <div className="results__item__header__location">
                {data.location}
              </div>
            </div>
          </button>
        </Link>

        <div className="results__item__company-info">
          <div className="results__item__company-info__column">
            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                IDNO:
              </p>
              <p className="results__item__company-info__column__item--value">
                {data.idno}
              </p>
            </div>

            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                Status
              </p>
              <button
                type="button"
                className="results__item__company-info__column__item--value"
              >
                ACTIV
              </button>
            </div>

            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                Date of establishment:
              </p>
              <p className="results__item__company-info__column__item--value">
                {data.creation_year}
              </p>
            </div>
            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                Varsta
              </p>
              <p className="results__item__company-info__column__item--value">
                {yearsOld}
              </p>
            </div>
          </div>
          <div className="results__item__company-info__column">
            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                Nr. by the employees:
              </p>
              <p className="results__item__company-info__column__item--value">
                {data.employees}
              </p>
            </div>
            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                Turnover:
              </p>
              <p className="results__item__company-info__column__item--value">
                {`${data.turnover} MDL`}
              </p>
            </div>
            <div className="results__item__company-info__column__item">
              <p className="results__item__company-info__column__item--key">
                Industry
              </p>
              <p className="results__item__company-info__column__item--value">
                {data.industry}
              </p>
            </div>
          </div>
          <div className="results__item__company-info__column--row">
            <div className="results__item__company-info__column__item--key">
              Contact
            </div>
            <div>
              <div className="results__item__company-info__column__item--value">
                <div className="results__item__company-info__column__item__results">
                  <img
                    className="results__item__company-info__column__item__icon"
                    src="../phone.png"
                    alt="phone"
                  />
                  <p>Phone</p>
                </div>
              </div>
              <div className="results__item__company-info__column__item--value">
                <div className="results__item__company-info__column__item__results">
                  <img
                    className="results__item__company-info__column__item__icon"
                    src="../mobile.png"
                    alt="phone"
                  />
                  <p>Mobile</p>
                </div>
              </div>
              <div className="results__item__company-info__column__item--value">
                <div className="results__item__company-info__column__item__results">
                  <img
                    className="results__item__company-info__column__item__icon"
                    src="../email.png"
                    alt="phone"
                  />
                  <p>Email</p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
