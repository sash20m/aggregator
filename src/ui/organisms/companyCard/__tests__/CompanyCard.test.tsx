import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { CompanyCard } from '../CompanyCard';

const Props = {
  data: {
    name: 'Comapny Name',
    slug: 'aridon',
    location: 'Chisinau',
    website: 'https://ebs-integrator.com',
    idno: '123456789',
    employees: '10-20',
    turnover: '999999',
    industry: 'IT',
    mobile: true,
    phone: true,
    email: true,
    creation_year: 2020,
  },
  key: 'Name',
};
afterEach(cleanup);

describe('comapnyCard', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <CompanyCard data={Props.data} key={Props.key} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should go to company page', async () => {
    const { getByTestId } = render(
      <CompanyCard data={Props.data} key={Props.key} />
    );

    expect(getByTestId('aridon')).toBeInTheDocument();
  });
});
