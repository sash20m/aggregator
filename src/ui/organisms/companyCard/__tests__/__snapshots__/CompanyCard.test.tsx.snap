// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comapnyCard should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="results"
  >
    <div
      class="results__item"
    >
      <button
        class="results__item__header"
        data-testid="aridon"
        type="button"
      >
        <img
          alt=""
          class="results__item__header__avatar"
          src="https://account.globaldatabase.com/logo/ebs-integrator.com/"
        />
        <div>
          <div
            class="results__item__header__name"
          >
            Comapny Name
          </div>
          <div
            class="results__item__header__location"
          >
            Chisinau
          </div>
        </div>
      </button>
      <div
        class="results__item__company-info"
      >
        <div
          class="results__item__company-info__column"
        >
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              IDNO:
            </p>
            <p
              class="results__item__company-info__column__item--value"
            >
              123456789
            </p>
          </div>
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              Status
            </p>
            <button
              class="results__item__company-info__column__item--value"
              type="button"
            >
              ACTIV
            </button>
          </div>
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              Date of establishment:
            </p>
            <p
              class="results__item__company-info__column__item--value"
            >
              2020
            </p>
          </div>
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              Varsta
            </p>
            <p
              class="results__item__company-info__column__item--value"
            >
              0
            </p>
          </div>
        </div>
        <div
          class="results__item__company-info__column"
        >
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              Nr. by the employees:
            </p>
            <p
              class="results__item__company-info__column__item--value"
            >
              10-20
            </p>
          </div>
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              Turnover:
            </p>
            <p
              class="results__item__company-info__column__item--value"
            >
              999999 MDL
            </p>
          </div>
          <div
            class="results__item__company-info__column__item"
          >
            <p
              class="results__item__company-info__column__item--key"
            >
              Industry
            </p>
            <p
              class="results__item__company-info__column__item--value"
            >
              IT
            </p>
          </div>
        </div>
        <div
          class="results__item__company-info__column--row"
        >
          <div
            class="results__item__company-info__column__item--key"
          >
            Contact
          </div>
          <div>
            <div
              class="results__item__company-info__column__item--value"
            >
              <div
                class="results__item__company-info__column__item__results"
              >
                <img
                  alt="phone"
                  class="results__item__company-info__column__item__icon"
                  src="../phone.png"
                />
                <p>
                  Phone
                </p>
              </div>
            </div>
            <div
              class="results__item__company-info__column__item--value"
            >
              <div
                class="results__item__company-info__column__item__results"
              >
                <img
                  alt="phone"
                  class="results__item__company-info__column__item__icon"
                  src="../mobile.png"
                />
                <p>
                  Mobile
                </p>
              </div>
            </div>
            <div
              class="results__item__company-info__column__item--value"
            >
              <div
                class="results__item__company-info__column__item__results"
              >
                <img
                  alt="phone"
                  class="results__item__company-info__column__item__icon"
                  src="../email.png"
                />
                <p>
                  Email
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
